example/src/main/java/io/netty/example/proxy/HexDumpProxyFrontendHandler.java

```diff
@@ -48,9 +49,9 @@ public void channelActive(ChannelHandlerContext ctx) {
          .channel(ctx.channel().getClass())
          .handler(new HexDumpProxyBackendHandler(inboundChannel))
          .option(ChannelOption.AUTO_READ, false);
-        ChannelFuture f = b.connect(remoteHost, remotePort);
-        outboundChannel = f.channel();
-        f.addListener((ChannelFutureListener) future -> {
+        outboundChannel = b.register().syncUninterruptibly().getNow();
```

Comment:
(from https://github.com/netty/netty/pull/11517#discussion_r678047564)

> To get the outboundChannel before connect completes.

I would suggest using the more idiomatic `ChannelInitializer` then:

```java
 .handler(new ChannelInitializer<>() {
     @Override
     protected void initChannel(Channel ch) {
         outboundChannel = ch;
         ch.pipeline().addLast(new HexDumpProxyBackendHandler(inboundChannel));
     }
 })
```

and preserve the old `connect()`.